packages/big-design/src/components/Tree/reducer.utils.ts

```diff
@@ -0,0 +1,181 @@
+import { Action } from './reducer';
+import { MapValues, TreeNodeId, TreeNodeProps, TreeProps, TreeState } from './types';
+
+export const initialize = <T>(nodes: TreeProps<T>['nodes'], radio: boolean): TreeState<T> => {
+  const initialState: TreeState<T> = {
+    nodeMap: new Map<TreeNodeId, MapValues>(),
+    expandedNodeIds: new Set(),
+    flattenedNodeIds: [],
+    focusedNode: null,
+    selectedValues: new Set(),
+    visibleNodeIds: [],
+  };
+
+  return recursiveInitialize(nodes, initialState, radio);
+};
+
+const recursiveInitialize = <T>(
+  nodes: TreeProps<T>['nodes'] | undefined,
+  state: TreeState<T>,
+  radio: boolean,
+): TreeState<T> => {
+  if (!nodes || nodes.length < 1) {
+    return state;
+  }
+
+  // Order of statements in this loop is important.
+  nodes.forEach((node) => {
+    state.flattenedNodeIds.push(node.id);
+
+    state.nodeMap.set(node.id, {
+      children: new Set(node.children?.map((child) => child.id)),
+      id: node.id,
+      parent: getParentId(state.nodeMap, node.id),
+    });
+
+    if (state.focusedNode === null) {
+      state.focusedNode = node.id;
+    }
+
+    if (node.expanded) {
+      state.expandedNodeIds.add(node.id);
+    }
+
+    const parent = state.nodeMap.get(node.id)?.parent;
```

Comment:
🍹 Can you do 
```js
const parent = getParentId(state.nodeMap, node.id);
```

Then use `parent` when you are setting the `nodeMap` and in the following statements.